generate.py

```diff
@@ -0,0 +1,116 @@
+from pathlib import Path
+
+from aind_data_schema_models.generators._base import (
+    _HarpDeviceTypeModel,
+    _ModalityModel,
+    _PlatformModel,
+    _SpeciesModel,
+    _MouseAnatomyModel,
+    _RegistryModel,
+)
+from aind_data_schema_models.generators.generators import (
+    GeneratorContext,
+    MappableReferenceField,
+    ModelGenerator,
+    ForwardClassReference,
+)
+from aind_data_schema_models.generators.parsers import csv_parser, get_who_am_i_list
+
+if __name__ == "__main__":
+
+    root = Path(__file__).parent / r"src/aind_data_schema_models/models"
+    target_folder = Path(r".\src\aind_data_schema_models\_generated")
+
+    platforms = ModelGenerator(
+        enum_like_class_name="_Platform",
+        parent_model_type=_PlatformModel,
+        discriminator="name",
+        data_source_identifier="platforms.csv",
+        parser=lambda: csv_parser(root / "platforms.csv"),
+    )
+
+    modalities = ModelGenerator(
+        enum_like_class_name="_Modality",
+        parent_model_type=_ModalityModel,
+        discriminator="name",
+        data_source_identifier="modalities.csv",
+        parser=lambda: csv_parser(root / "modalities.csv"),
+    )
+
+    harp_device_types = ModelGenerator(
+        enum_like_class_name="_HarpDeviceType",
+        parent_model_type=_HarpDeviceTypeModel,
+        discriminator="name",
+        data_source_identifier="https://raw.githubusercontent.com/harp-tech/protocol/97ded281bd1d0d7537f90ebf545d74cf8ba8805e/whoami.yml",  # noqa: E501
+        parser=lambda: get_who_am_i_list(
+            url="https://raw.githubusercontent.com/harp-tech/protocol/97ded281bd1d0d7537f90ebf545d74cf8ba8805e/whoami.yml"  # noqa: E501
```

Comment:
I wonder if we can separate this out into a different PR?

Reply:
Sounds good!